tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,229 @@
+import functools
+import operator
+import yaml
+
+import pytest
+from pytest_bdd import scenario, given, then, when, parsers
+
+import kubernetes.client
+from kubernetes.client import AppsV1Api
+from kubernetes.client.rest import ApiException
+
+from tests import utils
+
+
+# Fixtures {{{
+
+
+@pytest.fixture
+def k8s_appsv1_client(k8s_apiclient):
+    return AppsV1Api(api_client=k8s_apiclient)
+
+# }}}
+
+
+# Scenarios {{{
+
+
+@scenario('../features/service_configuration.feature',
+          'Propagation of Service Configurations to underlying Services')
+def test_service_config_propagation(host):
+    pass
+
+
+# }}}
+
+
+# Given {{{
+
+
+@given(parsers.parse(
+    "we have a '{name}' CSC in namespace '{namespace}' with "
+    "'{path}' equal to '{value}'"))
+def check_csc_configuration(
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+    csc_response = read_namespace_config_map(k8s_client, name, namespace)
+
+    assert csc_response, (
+        "No ConfigMap with name {} in namespace {} found".format(
+            name, namespace
+        )
+    )
+
+    csc_obj = yaml_load(csc_response, name, namespace)
+    response_value = traversed_dict(csc_obj, path)
+
+    assert int(value) == response_value, (
+        "Expected value {} for key {} in ConfigMap {} found in namespace {}, "
+        "got {}".format(value, path, name, namespace, response_value)
+    )
+
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we update '{name}' CSC in namespace '{namespace}' "
+    "'{path}' to '{value}'"))
+def update_service_configuration(
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+
+    full_csc = read_namespace_config_map(k8s_client, name, namespace)
+    csc_obj = yaml_load(full_csc, name, namespace)
+    csc_obj['spec']['deployment']['replicas'] = int(value)
```

Comment:
Nop, use `path` also here

Reply:
I also don't really like the hardcoded `int` here because we want to use this same function for others service configuration (that may be not int).

Maybe we should have 2 separate `when` one with quoted `{path}` for string and one without for int

Reply:
> Maybe we should have 2 separate `when` one with quoted `{path}` for string and one without for int

I think you can use a single `when`, and it can works fine if you use `"foo"` for string and `4` for integer.

`eval` is really bad, but [`literal_eval`](https://docs.python.org/3.6/library/ast.html#ast.literal_eval) ("Safely evaluate an expression node or a string containing a Python literal") is pretty cool.

```
>>> ast.literal_eval('3')
3
>>> ast.literal_eval('"foo"')
'foo'
```

Reply:
oh ok i didn't know this one, cool :+1: